spec/index.bs

```diff
@@ -11090,6 +11097,13 @@ dictionary GPURenderPassColorAttachment {
 
     - |renderViewDescriptor|.{{GPUTextureViewDescriptor/format}} must be a [=color renderable format=].
     - |this|.{{GPURenderPassColorAttachment/view}} must be a [$renderable texture view$].
+    - If |renderViewDescriptor|.{{GPUTextureViewDescriptor/dimension}} is {{GPUTextureViewDimension/"3d"}}:
+        - |this|.{{GPURenderPassColorAttachment/depthSlice}} must [=map/exist|be provided=] and must
+            be &lt; the [=GPUExtent3D/depthOrArrayLayers=] of the [=logical miplevel-specific texture extent=]
+            of the |renderTexture| [=subresource=] at [=mipmap level=] |renderViewDescriptor|.{{GPUTextureViewDescriptor/baseMipLevel}}.
+        - |this|.{{GPURenderPassColorAttachment/depthSlice}} has not been used earlier for same [=subresource=]
+            of the |renderTexture|.
```

Comment:
In addition to this, we also need a carveout that allows a single 3d subresource to be used as 'attachment' multiple times.

Unfortunately the way things are defined right now makes it difficult to express this well. I think we should replace this rule (since it's in a section that validates only one attachment at a time) and instead edit here:
https://github.com/gpuweb/gpuweb/blob/main/spec/index.bs#L9518-L9526

Here's my suggested change:
https://github.com/haoxli/gpuweb/compare/4251-render-to-3d-texture...kainino0x:gpuweb:render-to-3d-usage-scopes

Reply:
Thanks for your suggestion.